Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -205,142 +205,118 @@ extension NavigationMapView {
                 timer.invalidate()
                 return
             }
+            
             let newFractionTraveled = self.preFractionTraveled + traveledDifference * timePassedInMilliseconds.truncatingRemainder(dividingBy: 1000) / 1000
             
-            if let mainRouteLayerGradient = self.routeLineGradient(routeProgress.route, fractionTraveled: newFractionTraveled) {
-                self.mapView.style.updateLayer(id: mainRouteLayerIdentifier, type: LineLayer.self) { (lineLayer) in
-                    lineLayer.paint?.lineGradient = .expression(Expression.routeLineGradientExpression(mainRouteLayerGradient))
-                }
+            let congestionSegments = routeProgress.route.congestionFeatures(roadClassesWithOverriddenCongestionLevels: self.roadClassesWithOverriddenCongestionLevels)
+            let mainRouteLayerGradient = self.routeLineGradient(congestionSegments,
+                                                                fractionTraveled: newFractionTraveled)
```

Comment:
Calculations here cause significant performance issues on long routes. Improvements should be addressed in scope of another ticket.